buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -545,11 +562,65 @@ static int writeUtf8(AbstractByteBuf buffer, int writerIndex, CharSequence seq,
         return writerIndex - oldWriterIndex;
     }
 
+    /**
+     * Returns max bytes length of UTF8 character sequence of the given length.
+     */
+    public static int utf8MaxBytes(final int seqLength) {
+        return seqLength * MAX_BYTES_PER_CHAR_UTF8;
+    }
+
     /**
      * Returns max bytes length of UTF8 character sequence.
+     * <p>
+     * It behaves like {@link #utf8MaxBytes(int)} applied to {@code seq} {@link CharSequence#length()}.
      */
     public static int utf8MaxBytes(CharSequence seq) {
-        return seq.length() * MAX_BYTES_PER_CHAR_UTF8;
+        return utf8MaxBytes(seq.length());
+    }
+
+    /**
+     * Returns the exact bytes length of a well-formed UTF8 character sequence.
+     * <p>
+     * This method is producing the exact length only with well-formed UTF8 character sequences, otherwise
```

Comment:
s/UTF-8/UTF-16/ ?

Reply:
@ejona86 
Thanks!

> I found computing the "additional" bytes to be confusing

what do you mean with confusing? As a design choice or how I've implemented it?
It is mainly due to a simpler handling of integer overflow (rare TBH) and for inlining reasons optimizing around (what I'm expecting to be)  the common case

Reply:
> what do you mean with confusing?

I found it harder to check correctness, since it adds another operation (conceptually), instead of absolute numbers you have to do subtraction to get relative numbers. "And it is 2...? But it is 3 bytes in UTF-8. Doh! But it was one char in UTF-16." Or "isn't there a missing `++`? Doh. It's two chars in UTF-16." In this sort of conversion logic I am quite trained to think it terms of "X is Y bytes in UTF-8".

Reply:
Now I got it :) 
Yes it is less straightforward if compared to an absolute computation and TBH I could have implement it that way too (ie absolute encoded lenght): it is nice to see how different devs consider one way more intuitive than others...and is cool :)
If many devs will consider it less simple to be maintained than an absolute computation i will change it for sure :+1:

Reply:
@ejona86 
> s/UTF-8/UTF-16/ ?
I've used UTF8 as it is used in the already existing `ByteBufUtil`'s javadoc :
```
    /**
     * Returns max bytes length of UTF8 character sequence.
     * <p>
     * It behaves like {@link #utf8MaxBytes(int)} applied to {@code seq} {@link CharSequence#length()}.
     */
    public static int utf8MaxBytes(CharSequence seq) {
        return utf8MaxBytes(seq.length());
    }
```

Reply:
@franz1981, I wasn't talking about that first sentence. I was talking about the line I made the comment on:
> This method is producing the exact length only with well-formed UTF8 character sequences

The input is UTF-16 and is expected to be well-formed. The input isn't well-formed UTF-8.

Reply:
oh!!! You're pretty right, I've missed it!! thanks!